src/apps/KintoAppRegistry.sol

```diff
@@ -0,0 +1,271 @@
+        address parentContract,
+        address[] calldata childContracts,
+        uint256[4] calldata appLimits
+    ) external override {
+        require(appLimits.length == 4, "Invalid app limits");
+        require(msg.sender == _appMetadata[parentContract].developerWallet, "Only developer can update metadata");
+        _updateMetadata(_name, parentContract, childContracts, appLimits);
+        emit AppUpdated(parentContract, msg.sender, block.timestamp);
+    }
+
+    /**
+     * @dev Allows the app to request PII data
+     * @param app The name of the app
+     */
+    function enableDSA(address app) external override onlyOwner {
+        require(_appMetadata[app].dsaEnabled == false, "DSA already enabled");
+        _appMetadata[app].dsaEnabled = true;
+        emit AppDSAEnabled(app, block.timestamp);
+    }
+
+    /* ============ App Info Fetching ============ */
+
+    /**
+     * @dev Returns the metadata of the app
+     * @param _contract The address of the app
+     * @return The metadata of the app
+     */
+    function getAppMetadata(address _contract) external view override returns (IKintoAppRegistry.Metadata memory) {
+        address finalContract =
+            childToParentContract[_contract] != address(0) ? childToParentContract[_contract] : _contract;
+        return _appMetadata[finalContract];
+    }
+
+    /**
+     * @dev Returns the limits of the app
+     * @param _contract The address of the app
+     * @return The limits of the app
+     */
+    function getContractLimits(address _contract) external view override returns (uint256[4] memory) {
+        address finalContract =
+            childToParentContract[_contract] != address(0) ? childToParentContract[_contract] : _contract;
+        IKintoAppRegistry.Metadata memory metadata = _appMetadata[finalContract];
+        return [
+            metadata.rateLimitPeriod != 0 ? metadata.rateLimitPeriod : RATE_LIMIT_PERIOD,
+            metadata.rateLimitNumber != 0 ? metadata.rateLimitNumber : RATE_LIMIT_THRESHOLD,
+            metadata.gasLimitPeriod != 0 ? metadata.gasLimitPeriod : GAS_LIMIT_PERIOD,
+            metadata.gasLimitCost != 0 ? metadata.gasLimitCost : GAS_LIMIT_THRESHOLD
+        ];
+    }
+
+    /**
+     * @dev Returns whether a contract is sponsored by an app
+     * @param _app The address of the app
+     * @param _contract The address of the contract
+     * @return bool true or false
+     */
+    function isContractSponsored(address _app, address _contract) external view override returns (bool) {
+        return _contract == _app || childToParentContract[_contract] == _app || _appSponsoredContracts[_app][_contract];
+    }
+
+    /**
+     * @dev Returns the contract that sponsors a contract
+     * @param _contract The address of the contract
+     * @return The address of the contract that sponsors the contract
+     */
+    function getContractSponsor(address _contract) external view override returns (address) {
```

Comment:
```suggestion
    function getSponsor(address _contract) external view override returns (address) {
```

Reply:
what happens on the following situation:
- App 1 with contract 0x123 registers with children 0x111, 0x222 and 0x333
- App 2 with contract 0x222 registers with children 0x444, 0x555 and 0x666

- If I call `getContractSponsor(0x111)` returns 0x123
- If I call `getContractSponsor(0x222)` it will return 0x222, is that the intended behaviour?

Reply:
yes